protocol/communities/manager.go

```diff
@@ -1399,7 +1407,37 @@ func (m *Manager) HandleCommunityAdminEvent(signer *ecdsa.PublicKey, adminEvent
 		return nil, err
 	}
 
-	return m.handleCommunityDescriptionMessageCommon(community, patchedCommDescr, rawMessage)
+	response, err := m.handleCommunityDescriptionMessageCommon(community, patchedCommDescr, rawMessage)
+	if err != nil {
+		return nil, err
+	}
+
+	if adminEvent.CommunityConfig != nil {
+		settings, err := m.persistence.GetCommunitySettingsByID(community.ID())
+		if err != nil {
+			return nil, err
+		}
+
+		if settings == nil {
+			settings = &CommunitySettings{
+				CommunityID:                  string(adminEvent.CommunityId),
+				HistoryArchiveSupportEnabled: adminEvent.CommunityConfig.HistoryArchiveSupportEnabled,
+				Clock:                        adminEvent.Clock,
+			}
+		}
+
+		if adminEvent.Clock > settings.Clock {
+			settings.HistoryArchiveSupportEnabled = adminEvent.CommunityConfig.HistoryArchiveSupportEnabled
+			settings.Clock = adminEvent.Clock
+		}
+
+		err = m.persistence.SaveCommunitySettings(*settings)
+		if err != nil {
+			return nil, err
+		}
+		response.CommunitySettings = settings
```

Comment:
@mprakhov @0x-r4bbit @jrainville 
Here I'm trying to sync `HistoryArchiveSupportEnabled` between community admins and it's coming out pretty messy. Is there a reason why we keep this setting separate in CommunitySettings but not in CommunityDescription?

On the other hand, can we use `SyncCommunitySettings` not only with other devices, but but with other admins?

Reply:
As i see [here](https://github.com/status-im/status-desktop/blob/fix/issue-11089-community-toggling-history-service-as-admin-is-not-reflected-to-the-owner-node/src/app_service/service/node_configuration/service.nim#L59) this setting make sense only for desktop owner node